FaceDetectionServer/REST-API.md

```diff
@@ -0,0 +1,240 @@
+{"success": "false", "server_processing_time": "0.030"}
+```
+
+## Face Recognition
+`/recognizer/predict/`
+
+Used to send a face image to the server and get back a set of coordinates for the recognized face, along with a "subject" name value if the face is recognized against the current trained data.
+
+_Note: This call is limited to a single face. At training time, no images are added to the set if more than one face is detected. If more than one face is present in an image for this call, prediction will only be performed on the first face detected._
+
+### POST
+```
+http://<hostname>:8008/recognizer/predict/
+```
+
+#### Parameters
+| Parameter | Description |
+| - | - |
+| image | The face image. |
+
+#### Return
+JSON string with a "success" value of "true" or "false", and if successful, the following JSON values:
+
+| Attribute | Description |
+| - | - |
+| success | "True" if face recognized, otherwise "false". |
+| subject | Name of detected face. |
+| confidence | Level of confidence. Actually a measure of distance the features are from matching a given subject. Lower is better, with 0 being a perfect match. |
+| rect | Coordinates of detected face. |
+| server_processing_time | Time in milliseconds that it took the server to process this image. |
+
+_Note: This call will return subject names even with very poor (high distance) confidence values. It is up to the caller to decide what it considers a good prediction. For instance in the [MobiledgeX SDK Demo app](https://play.google.com/store/apps/details?id=com.mobiledgex.sdkdemo), the *RECOGNITION_CONFIDENCE_THRESHOLD* value is 105, which means that any image with a confidence value higher than that will be shown with a subject name of "Unknown"._
+
+#### Examples:
+##### Face detected and recognized:
+```
+curl facedetection.defaultedge.mobiledgex.net:8008/recognizer/predict/ -F "image=@Bruce.jpg"
+{"success": "true", "subject": "Bruce Armstrong", "confidence": "71.850", "server_processing_time": "48.218, "rect": [73, 78, 146, 151]}
+```
+
+##### Face not detected:
+```
+curl facedetection.defaultedge.mobiledgex.net:8008/recognizer/predict/ -F "image=@single_pixel.png"
+{"success": "false", "server_processing_time": "0.082"}
+```
+##### Training data update in progress:
+```
+curl facedetection.defaultedge.mobiledgex.net:8008/recognizer/predict/ -F "image=@Bruce.jpg"
+<HTTP/1.1 503 Service Unavailable>
+Training data update in progress
+```
+If this state is encountered, it should only last a few seconds.
+
+## Pose Detection
+`/openpose/detect`
+
+Used to send an image with one or more human bodies to the server and get back a set of coordinates for any detected poses. In the our implementation, pose detection requires a GPU in order to run. If the server executing this call does not have GPU support, a `501 Not Implemented` response will be returned along with the message "OpenPose not supported on this server".
```

Comment:
It would be helpful to add a short definition for a pose. Something like : 

"where a pose is a list of bones corresponding to a human body"